src/main/java/com/thevoxelbox/voxelsniper/brush/GenerateTreeBrush.java

```diff
@@ -223,48 +182,46 @@ private void rootCreate(final int xDirection, final int zDirection) {
 
                 // If not solid then...
                 // Save for undo function
-                if (this.getBlockMaterialAt(blockPositionX, blockPositionY, blockPositionZ) != woodMaterial) {
-                    this.undo.put(this.clampY(blockPositionX, blockPositionY, blockPositionZ));
+                if (this.getBlockMaterialAt(location) != woodMaterial) {
+                    this.undo.put(location.getClampedBlock());
 
                     // Place log block.
-                    this.clampY(blockPositionX, blockPositionY, blockPositionZ).setBlockData(woodMaterial.createBlockData(), false);
+                    location.setBlockData(woodMaterial.createBlockData(), false);
                 } else {
                     // If solid then...
                     // End loop
                     break;
                 }
-
+                List<Material> blocks = Arrays.asList(Material.WATER, Material.SNOW, Material.OAK_LOG, Material.BIRCH_LOG, Material.ACACIA_LOG, Material.DARK_OAK_LOG, Material.SPRUCE_LOG, Material.JUNGLE_LOG);
```

Comment:
Not so sure what to do with this

Reply:
From the comment, it looks like this is supposed to check if the block is solid? Would Material#isSolid work here?
Also: how are water and snow considered solid blocks 😮

Reply:
Going to make this a seperate issue

Reply:
https://github.com/KevinDaGame/VoxelSniper/issues/32